app/lib/common/providers/notifiers/chat_notifiers.dart

```diff
@@ -204,6 +199,24 @@ class ChatRoomsListNotifier extends StateNotifier<List<Convo>> {
         break;
     }
   }
+
+  @override
+  List<Convo> build() {
+    _providerSubscription = ref.listen<AsyncValue<Client?>>(
+      alwaysClientProvider,
+      (AsyncValue<Client?>? oldVal, AsyncValue<Client?> newVal) {
+        final client = newVal.valueOrNull;
+        if (client == null) {
+          // we don't care for not having a proper client yet
+          return;
+        }
+        _reset(client);
+      },
+      fireImmediately: true,
+    );
+    ref.onDispose(() => _providerSubscription.close());
```

Comment:
main pattern used to `listen` to the client now from within the other `Notifiers`. Logic otherwise is mainly the same, just some more better cleaning up procedures added.

Reply:
only _once_ a client is found, do we actually call the inner function that's what this code does. at build there might not be one yet (as that provider is async).